src/probnum/problems/_problems.py

```diff
@@ -1 +1,202 @@
-"""Definitions of problems solved by probabilistic numerical methods."""
+"""Definitions of problems currently solved by probabilistic numerical methods.
+
+If a problem is not listed here, we have not implemented a probabilistic
+numerical method for it yet.
+"""
+
+import dataclasses
+import typing
+
+import numpy as np
+import scipy.sparse
+
+import probnum.filtsmooth as pnfs
+import probnum.linops as pnlo
+import probnum.random_variables as pnrv
+import probnum.type as pntp
+
+
+@dataclasses.dataclass
+class RegressionProblem:
+    r"""Regression problem.
+
+    Fit a stochastic process to data, given a likelihood (realised by a :obj:`Transition`).
+    Solved by Kalman filtering and smoothing in :mod:`probnum.filtsmooth`.
+
+    Parameters
+    ----------
+    observations
+        Observations of the latent process.
+    locations
+        Grid-points on which the observations were taken.
+    likelihood
+        Likelihood of the observations; that is, relation between the latent process and the observed values.
+        Encodes for example noise.
+    solution
```

Comment:
I don't really have a strong opinion on naming. I guess alternatives would be "true_solution", "analytical" (which is how SciML does it), "analytical_solution", "sol", or others. If you think one of them is strictly better than others, please voice your opinion. I guess the only thing to take into account is to communicate that calling "problem.solution" will **not** solve your problem, therefore something overly specific or even tedious to call might not be a bad idea.

Reply:
I think `solution` is a good general naming. If `solution=None` and therefore not set, then one has to call `ProbabilisticNumericalMethod.solve(problem)` to obtain an approximate one.